webpack/assets/javascripts/react_app/common/MountingService.js

```diff
@@ -7,21 +7,49 @@ import componentRegistry from '../components/componentRegistry';
 
 export { default as registerReducer } from '../redux/reducers/registerReducer';
 
-const mountedNodes = [];
+/**
+ * mounted-nodes cache
+ */
+let mountedNodes = [];
 
 // In order to support turbolinks with react, need to
 // unmount all root components before turbolinks do the unload
 // TODO: remove it when migrating into (webpacker-react or react-rails)
-document.addEventListener('page:before-unload', () => {
-  let node = mountedNodes.shift();
+document.addEventListener('page:before-unload', () => unmountComponents());
+
+const unmountComponentsByNodes = (...nodes) => {
+  let node = nodes.shift();
 
   while (node) {
     ReactDOM.unmountComponentAtNode(node);
-    node = mountedNodes.shift();
+    node = nodes.shift();
   }
-});
+};
+
+/**
+ * Will unmount all react-components from the dom
+ * @param  {string} selector unmount component inside the selector
+ * @return {number}          amount of unmounted nodes
+ */
```

Comment:
Just curious, when did we start adding comments to functions?

Reply:
I did, but only when it's core related.
I plan to do some experiment later and generate docs from it.